kratos/geometries/coupling_geometry.h

```diff
@@ -0,0 +1,357 @@
+    }
+
+    explicit CouplingGeometry(const PointsArrayType& ThisPoints)
+        : BaseType()
+    {
+    }
+
+    /**
+     * Copy constructor.
+     * Construct this geometry as a copy of given geometry.
+     *
+     * @note This copy constructor does not copy the points and new
+     * geometry shares points with given source geometry. It is
+     * obvious that any change to this new geometry's point affect
+     * source geometry's points too.
+     */
+    CouplingGeometry( CouplingGeometry const& rOther )
+        : BaseType( rOther )
+    {
+    }
+
+    /**
+     * Copy constructor from a geometry with other point type.
+     * Construct this geometry as a copy of given geometry which
+     * has different type of points. The given goemetry's
+     * TOtherPointType* must be implicity convertible to this
+     * geometry PointType.
+     *
+     * @note This copy constructor does not copy the points and new
+     * geometry shares points with given source geometry. It is
+     * obvious that any change to this new geometry's point affect
+     * source geometry's points too.
+     */
+    template<class TOtherPointType> explicit CouplingGeometry(
+        CouplingGeometry<TOtherPointType> const& rOther )
+        : BaseType( rOther )
+    {
+    }
+
+    /**
+     * Destructor. Does nothing!!!
+     */
+    ~CouplingGeometry() override {}
+
+
+    GeometryData::KratosGeometryFamily GetGeometryFamily() const override
+    {
+        return GeometryData::Kratos_generic_family;
+    }
+
+    GeometryData::KratosGeometryType GetGeometryType() const override
+    {
+        return GeometryData::Kratos_generic_type;
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * Assignment operator.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    CouplingGeometry& operator=( const CouplingGeometry& rOther )
+    {
+        BaseType::operator=( rOther );
+        return *this;
+    }
+
+    /**
+     * @brief Assignment operator for geometries with different point type.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    template<class TOtherPointType>
+    CouplingGeometry& operator=(
+        CouplingGeometry<TOtherPointType> const & rOther )
+    {
+        BaseType::operator=( rOther );
+        return *this;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    typename BaseType::Pointer Create(
+        PointsArrayType const& ThisPoints ) const override
+    {
+        return typename BaseType::Pointer(
```

Comment:
```suggestion
        return Kratos::make_shared<CouplingGeometry>(ThisPoints);
```

Reply:
Are you sure that this is the same? BaseType is of type: Geometry<PointType>

Reply:
the fct returns `BaseType::Pointer`, but you create an object of type `CouplingGeometry`

Hence I am quite sure it will work, but please try if you are unsure